ex04.ex

```diff
@@ -70,12 +81,25 @@ defmodule Ex04 do
       iex> Ex04.even_odd [ 1, 2, 3, 4, 5 ]
       { [ 2, 4],  [ 1, 3, 5 ] }
 
+      iex> Ex04.even_odd [ 1, 7, 2, 18, 3, 4, 5 ]
+      { [ 2, 18, 4],  [ 1, 7, 3, 5 ] }
+
   Hint: you're taking a list and converting it into something else. What function
   helps you do that. And, if you use that function, what does it return? That
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+  def even_odd (list) do
```

Comment:
## -2

This function duplicates both code and the processing of the list. Think about how to simplify things by factoring out functions

~~~ elixir
  def even_odd(list) do
    list
    |> reverse()
    |> reduce({ [], [] }, &partition_even_odd/2)
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}
~~~